src/lib/utils/api/alchemy.ts

```diff
@@ -0,0 +1,63 @@
+import { getFlashMintLeveragedContract } from '@indexcoop/flash-mint-sdk'
+import { Alchemy, AssetTransfersCategory, Network } from 'alchemy-sdk'
+import { Address, zeroAddress } from 'viem'
+import * as chains from 'viem/chains'
+
+const apiKey = process.env.NEXT_PUBLIC_ALCHEMY_ID
+
+const AlchemyApi = {
+  [chains.arbitrum.id]: new Alchemy({
+    apiKey,
+    network: Network.ARB_MAINNET,
+  }),
+  [chains.base.id]: new Alchemy({
+    apiKey,
+    network: Network.BASE_MAINNET,
+  }),
+}
+
+type SupportedChainId = keyof typeof AlchemyApi
+
+export const fetchTokenTransfers = async (
+  user?: Address,
+  contractAddresses?: Address[],
+  chainId?: number,
+) => {
+  if (
+    !user ||
+    !contractAddresses?.length ||
+    !chainId ||
+    chainId in AlchemyApi === false
+  )
+    return []
+
+  const client = AlchemyApi[chainId as SupportedChainId]
+
+  const flashMintContract = getFlashMintLeveragedContract(
```

Comment:
To be save always use `getFlashMintLeveragedContractForToken` to get the right leveraged contract for a token: https://github.com/IndexCoop/flash-mint-sdk/blob/main/src/utils/contracts.ts#L145. Otherwise, you might get the wrong contract. We should probably deprecate the other util functions in the SDK some time.

Reply:
After testing this doesn't seem to be working as intended. I don't get back the close positions with `getFlashMintLeveragedContractForToken` but I do with `getIndexFlashMintLeveragedExtendedContract`